lib/event_listeners.js

```diff
@@ -293,6 +293,18 @@ AWS.EventListeners = {
         var api = service.api;
         var operationName = resp.request.operation;
         var operation = api.operations[operationName] || {};
+        var errorCodeMapping = api.errorCodeMapping;
+
+        if (errorCodeMapping) {
+          request.on('error', function (err, response) {
+            if (response.error && response.error.code) {
+              var mapping = errorCodeMapping[response.error.code];
+              if (mapping) {
+                response.error.code = mapping.code;
+              }
+            }
+          },true);
+        }
```

Comment:
This should be equivalent to:

```js
add('Error', 'error', function MAP_ERROR_CODE(err, response) {
  // same code
}
```
outside of the `SEND()`.

Reply:
Sounds good!